### ISSUE TITLE
tB ListBox.AddItem Method Signature is Incompatible with VB6 ListBox.AddItem Method Signature

### ISSUE THREAD

**Issue by jpbro** [WaynePhillipsEA/twinbasic#776]
**Describe the bug**
The twinBasic AddItem method's Index parameter is required, but the VB6 Index parameter is optional. This results in errors when migrating VB6 code.

**Expected behavior**
I expect the tB ListBox.AddItem method signature to match the VB6 ListBox.AddItem method signature.

**Comment by WaynePhillipsEA** [comment]
Can't seem to reproduce this.  The AddItem Index parameter appears to already be optional.

**Comment by jpbro** [comment]
Interesting, I think it might be a weird symptom of another problem in the other project that I emailed you. The following code in that project:

```
Function EnumFontFamProc(lpNLF As LOGFONT, lpNTM As NEWTEXTMETRIC, ByVal FontType As Long, lParam As ListBox) As Long
   Dim FaceName As String
   
   FaceName = StrConv(lpNLF.lfFaceName, vbUnicode)
   FaceName = Left$(FaceName, InStr(FaceName, vbNullChar) - 1)
   lParam.AddItem FaceName
   EnumFontFamProc = 1
End Function
```

Shows this error on the `lParam.AddItem FaceName` line:

![image](https://user-images.githubusercontent.com/5287585/161428519-dbfd8bd7-d4b6-4716-988b-c8829ad1a79b.png)

And the AddItem method signature popup shows that Index is not optional:

![image](https://user-images.githubusercontent.com/5287585/161428569-da0c18ef-789d-4e6b-83e1-48b97fd40e50.png)

I can't reproduce this issue in a new Standard EXE project, so I suspect there is some other code in my larger project that is causing the problem.

**Comment by jpbro** [comment]
One thing I notice is that in a new Standard EXE project, the popup window shows that the AddItem method is in `_ListBox`:

![image](https://user-images.githubusercontent.com/5287585/161429188-f6f012eb-da5c-4853-a86a-c39f4f165faf.png)

But in my larger project it says that it is in `VB._ListBox`.

![image](https://user-images.githubusercontent.com/5287585/161429261-5cb89fdc-8872-4614-aebc-dc0ec287d464.png)

Maybe a clue?

**Comment by WaynePhillipsEA** [comment]
> Maybe a clue?

Yup... the standard EXE project will be using the WinNativeControls package, that includes the tB ListBox definition, whilst your other project won't be referencing that package.  One of your references will be indirectly referencing the real VB6 VB type library.

**Comment by WaynePhillipsEA** [comment]
The real VB6 type library stupidly does not include the OPTIONAL attribute of the parameter:

```
[helpstring("Adds an item to a Listbox or ComboBox control or a row to a Grid control."), helpcontext(0x000dfa84)]
    HRESULT _stdcall AddItem(
                    [in] BSTR Item, 
                    [in] VARIANT Index);
```

So, tB will have to patch that up when it encounters the real VB6 type library.

**Comment by jpbro** [comment]
Ugg, well at least we can blame this one on @Microsoft ;)
